contracts/axelar-gas-service/src/contract.rs

```diff
@@ -0,0 +1,120 @@
+use soroban_sdk::{
+    contract, contractimpl, panic_with_error, token, Address, Bytes, Env, String, U256,
+};
+
+use axelar_soroban_std::types::{Hash, TokenDetails};
+
+use crate::storage_types::DataKey;
+use crate::{error::Error, event};
+use axelar_soroban_interfaces::axelar_gas_service::AxelarGasServiceInterface;
+
+#[contract]
+pub struct AxelarGasService;
+
+#[contractimpl]
+impl AxelarGasService {
+    pub fn initialize(env: Env, gas_collector: Address) {
+        if env
+            .storage()
+            .instance()
+            .get(&DataKey::Initialized)
+            .unwrap_or(false)
+        {
+            panic!("Already initialized");
+        }
+
+        env.storage().instance().set(&DataKey::Initialized, &true);
+
+        env.storage()
+            .instance()
+            .set(&DataKey::GasCollector, &gas_collector);
+    }
+}
+
+#[contractimpl]
+impl AxelarGasServiceInterface for AxelarGasService {
+    fn pay_gas_for_contract_call(
+        env: Env,
+        sender: Address,
+        destination_chain: String,
+        destination_address: String,
+        payload: Bytes,
+        refund_address: Address,
+        token_details: TokenDetails,
```

Comment:
```suggestion
        token: Token,
```
TokenDetails sounds like it contains name, symbol etc as well